src/DataStructures/Mesh.cpp

```diff
@@ -12,21 +12,35 @@ template <size_t Dim>
 // clang-tidy: incorrectly reported redundancy in template expression
 template <size_t N, Requires<(N > 0 and N == Dim)>>  // NOLINT
 Mesh<Dim - 1> Mesh<Dim>::slice_away(const size_t d) const noexcept {
-  std::array<size_t, Dim - 1> slice_extents{};
-  std::array<Spectral::Basis, Dim - 1> slice_bases{};
-  std::array<Spectral::Quadrature, Dim - 1> slice_quadratures{};
-  for (size_t i = 0; i < Dim; ++i) {
-    if (i < d) {
-      gsl::at(slice_extents, i) = gsl::at(extents_.indices(), i);
-      gsl::at(slice_bases, i) = gsl::at(bases_, i);
-      gsl::at(slice_quadratures, i) = gsl::at(quadratures_, i);
-    } else if (i > d) {
-      gsl::at(slice_extents, i - 1) = gsl::at(extents_.indices(), i);
-      gsl::at(slice_bases, i - 1) = gsl::at(bases_, i);
-      gsl::at(slice_quadratures, i - 1) = gsl::at(quadratures_, i);
-    }
+  ASSERT(d < Dim, "Tried to slice away non-existing dimension "
+                      << d << " of " << Dim << "-dimensional mesh.");
+  std::array<size_t, Dim - 1> dims;
+  for (size_t dim = 0; dim < d; dim++) {
+    gsl::at(dims, dim) = dim;
```

Comment:
This just constructs an array with the numbers `0...Dim-1` with `d` omitted. Any suggestions on how to do this better?

Reply:
We have a function to remove an element from an array, so you could create one with all the values and then remove one of them.  I'm not sure whether that is an improvement.

Reply:
If you are referring to `all_but_specified_element_of` in `StdArrayHelpers.hpp`, that doesn't work here because `d` is a runtime value. Keeping the implementation for now then.

Reply:
Oops, I missed that.  Leaving as is sounds good to me then.